src/htm/algorithms/SpatialPooler.cpp

```diff
@@ -866,22 +902,22 @@ void SpatialPooler::inhibitColumnsLocal_(const vector<Real> &overlaps,
 
 
       if (wrapAround_) {
-         numNeighbors = 0;  // In wrapAround, number of neighbors to be considered is solely a function of the inhibition radius, 
-	 // ... the number of dimensions, and of the size of each of those dimenion
-         UInt predN = 1;
-	 const UInt diam = 2*inhibitionRadius_ + 1; //the inh radius can change, that's why we recompute here
-         for (const auto dim : columnDimensions_) {
-           predN *= std::min(diam, dim);
-         }
-         predN -= 1;
-         numNeighbors = predN;
-         const UInt numActive_wrap = static_cast<UInt>(0.5f + (density * (numNeighbors + 1)));
-
-        for(auto neighbor: WrappingNeighborhood(column, inhibitionRadius_,columnDimensions_)) { //TODO if we don't change inh radius (changes only every isUpdateRound()),
+        numNeighbors = 0;  // In wrapAround, number of neighbors to be cons     idered is solely a function of the inhibition radius, the number of dimensi     ons, and of the size of each of those dimenion
+        numNeighbors = wrapAroundNeighbors_;
+
+        const UInt numActive_wrap = (UInt)(0.5f + (density * (numNeighbors + 1)));
+	 
+	const UInt mapOffset = column * (numNeighbors+1);
```

Comment:
I'd like to shield off this logic of indexing in the map (of cached values) from the SP. Ideally later we extend Neighbourhood() to have param `preCache=true` and do all of this internally.

Reply:
So it should be done in such a way that we'd still use the same (auto neighbor: WrappingNeighborhood) logic, is that it?

Reply:
I think we'd like the data `const vector<SynapseIdx>[][]& mapNeighbors` accessible from WrappingNeighborhood and filled in constructor there. 
Used as
```
vector& neighbors = map[radius][col];
```

Reply:
I'll try. This would probably the hardest change to make, I think(?).

Reply:
yes, but I think it can be done quite nicely. (I haven't looked into the details, but this would be my high-level idea:) 
- add a new class CachedWrappingNeighborhood:
```
CachedWrappingNeighborhood(Neighborhood): 
constructor(rangeCenter, rangeRadius, dimColumns) {
  //this takes a lot of time, once
  for r in rangeRadius {
    for col in rangeCenter {
      hood = WrappingNeighborhood(col, r, colDimensions);
       vector locals={};
      for( neighbor : hood) {
        locals.append(neighbor); 
      }
      this->cachedHood[r][col] = locals;
  }}
}

```